src/code/z_play.c

```diff
@@ -503,7 +503,7 @@ void Play_Update(PlayState* this) {
             }
         }
 
-        if (this->transitionMode) { // != TRANS_MODE_OFF
+        if ((u32)this->transitionMode != TRANS_MODE_OFF) {
             switch (this->transitionMode) {
```

Comment:
not too sure about these. Id argue this is definitely turning it from a real match into a fake match. Its way more likely that the original devs wrote these kinds of conditions in a boolean style instead of casting to u32.
But casting these to be able to write out the whole condition could be argued as being more readable. So it can go either way

Reply:
to elaborate on my reasoning, the case doesnt really achieve anything or have any purpose. I dont think its likely a dev actually wrote that. Whereas not writing out != 0 is a common styling choice that programmers choose to use, even on non-booleans (even though our project specific styling preferences dont like to do it)

Reply:
I do understand that it does look a little more "fakey", but the motivation for doing this was mainly for the enums that will come into play later. In this particular case in `z_play`, either is fine and you can use the enum off to the side. It becomes nicer though when this is done in a bigger `if` block. For example:
```c
    if ((!pauseCtx->unk_1E4 || (pauseCtx->unk_1E4 == 5) || (pauseCtx->unk_1E4 == 8)) &&
```
to
```c
    if ((((u32)pauseCtx->unk_1E4 == 0) || (pauseCtx->unk_1E4 == 5) || (pauseCtx->unk_1E4 == 8)) &&
```
which using Dragorns WIP, would be something like:
```c
if ((((u32)pauseCtx->mainState == PAUSE_MAIN_STATE_IDLE) || (pauseCtx->mainState == PAUSE_MAIN_STATE_SONG_PROMPT) || (pauseCtx->mainState == PAUSE_MAIN_STATE_IDLE_CURSOR_ON_SONG)) &&
```
which looks a bit better than `!pauseCtx->mainState` inside a big set of conditionals. 

I also feel it reads more naturally with the `== 0` + fake cast, then the `!`, but that's also just me.

That's the motivation for changing it, but I don't feel too strongly about it and can always change it back

Reply:
yea even in the case of the pause state there I would argue that casting and using an enum is completely fake, and the dev used `!pauseCtx->unk_1E4` even though its not a boolean.
But yeah, more opinions needed on if we should still go this route anyway for the sake of using the enums directly in the code

Reply:
Yeah, I also completely agree that the og devs most likely had the `!pause->unk_1E4`, and that this is taking a creative liberty to deviate from that using a minor fake match to something that's clearer and easier to read (for me at least). So I still like how I've changed it, but will definitely change back if others prefer the og devs way or don't like the (u32) cast in there

Reply:
I like the cast hack

Is it that important to stay close to what was 99.9% likely the original source? We take other liberties elsewhere abusing fake matches, many without knowing probably, like with a bunch of macros I'd wager, or booleans in some cases

That said I can live without it if you really don't want it

Reply:
will it match if e.g. `PAUSE_MAIN_STATE_IDLE` is defined as `0u`?

Reply:
`(pauseCtx->unk_1E4 == 0u)` does not match

Reply:
I still like having this fake change when there's an enum that can be used for 0, but would be okay with just the `!var` when its' just checking against 0. But not sure if we've reached a consensus on this or not

Reply:
@fig02 Do you still have strong feelings against this OK hack?

Reply:
I am still not a fan yeah.
Asked @Roman971  to weigh in a while ago, and he must have forgot about it. But if he is also ok with it ill concede.

Reply:
I'm personally not a huge fan either, and I think we should generally avoid using fake tricks like this since it's debatable and not a clear improvement (same reason I'm not a fan of the 1d array cast we're doing with `sSetupDL` for instance).
That being said, this specific case is probably okay when there is an enum involved, since it can allow the first enum value to not be "0" without breaking stuff. So I think I'm fine with it

Reply:
alright, I will concede then